packages/openapi/api/actions.yaml

```diff
@@ -0,0 +1,84 @@
+openapi: '3.0.3'
+info:
+  title: 'Sofie User Actions API'
+  description: 'The Sofie User Actions API provides paths to allow a device to control a Sofie and query health information'
+  version: '0.0.1'
+  license:
+    name: 'MIT License'
+    url: 'http://opensource.org/licenses/MIT'
+servers:
+  - url: http://localhost:3000/api/v1.0
+    description: Initial release of UserActions API - http - for use during development only!
+tags:
+  - name: userActions
+    description: Provides the ability to control the Sofie application
+paths:
```

Comment:
Regarding all api-endpoints that returns lists of strings, for example `GET /devices`, `GET /playlists`, `GET /studios` etc.

I think they should instead return a list of objects: `{id: string}[]`, this makes it possible to extend the endpoints with more data at a later time without breaking the API. (an example is when extending to `{id: string, name: string}[]`)

Reply:
This is a good point. This actually made me think: should we rather wrap the collections into a wrapper object? Something like?
```json
{
"devices": [
{ "id": "123" },
{ "id": "345" }
]
}
```

A wrapper object would allow us to potentially also provide metadata about _the data collection_ in a forwards-compatible way - such as cursors for paging, if the need ever arises.

Reply:
Scratch what I wrote. I've played with this definition in: [ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/evs-broadcast/sofie-core/draft/sofie-api/packages/openapi/api/actions.yaml#tag/devices/operation/device) (also, is there a reason why this is using Swagger - which does require JRE - as opposed to a nodejs-based tool?), and now I see that there already is a wrapper object around the response, so that could carry metadata. I think @nytamin 's point still stands, though. Returning an Array of Objects would be better than an Array of Strings, for future proofing. One could imagine for example adding a `$href` property to the objects with links to `/devices/123`.